### ISSUE TITLE
Shops with multiple inventory pages require "_" command to enter

### ISSUE THREAD

**Issue by angband-import** [angband/angband#744]
**Reported by ekolis on 5 Jul 2007 20:25 UTC**
If I try to enter a shop with multiple pages of inventory, the screen just flickers and I don't get in, but if I press the "_" key I can get in. Just a bit of a nuisance!

**Comment by angband-import** [comment]
**Attachment from takkaria on 12 Jul 2007 14:25 UTC**
Reproduces issue for a reporter
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_202_GIT_744/1001.Spook

**Comment by angband-import** [comment]
**Modified by takkaria on 5 Jul 2007 22:24 UTC**

**Comment by angband-import** [comment]
**Modified by takkaria on 7 Jul 2007 20:30 UTC**

**Comment by angband-import** [comment]
**Modified by takkaria on 8 Jul 2007 14:00 UTC**

**Comment by angband-import** [comment]
**Modified by takkaria on 8 Jul 2007 20:46 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 12 Jul 2007 11:47 UTC**
Reported on SuSE Linux 10 and Xubuntu Feisty Fawn.

**Comment by angband-import** [comment]
**Comment by anonymous on 15 Jul 2007 11:30 UTC**
This is the fix (in case I accidentally trash it before svn comes back up):
```
Index: src/util.c
===================================================================
--- src/util.c  (revision 345)
+++ src/util.c  (working copy)
\@\@ -3222,6 +3222,7 \@\@
 
                        /* Use auto-command */
                        ke.key = (char)p_ptr->command_new;
+                       ke.type = EVT_KBRD;
 
                        /* Forget it */
                        p_ptr->command_new = 0;
```

**Comment by angband-import** [comment]
**Comment by ajps on 18 Jul 2007 21:35 UTC**
Fixed in [350] merged in [5dfe994] (SVN r399)
